### PR TITLE
added advertiseFrom field to the channel and annotatioons

### PR DESCRIPTION
**Pull request by Spartakovic**

### CODE REVIEW

**Comment by Dalamar42** [comment]
You need to also update:
- the Deer version of the model
- the code the transforms the Owl model to the Deer model
- the protocol buffers serialisation and deserialisation to ensure this is stored in C*

**Comment by Dalamar42** [comment]
LGTM

File: atlas-api/src/main/java/org/atlasapi/output/annotation/ChannelAdvertisedAnnotation.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+public class ChannelAdvertisedAnnotation extends OutputAnnotation<Channel> {
+    private final ChannelWriter channelWriter;
+
+    public ChannelAdvertisedAnnotation(ChannelWriter channelWriter) {
+        this.channelWriter = checkNotNull(channelWriter);
+    }
+
+    @Override
+    public void write(Channel entity, FieldWriter format, OutputContext ctxt) throws IOException {
```
Comment (tfm) [inline atlas-api/src/main/java/org/atlasapi/output/annotation/ChannelAdvertisedAnnotation.java]: These needs to honour the existing logic for channel output, too, which is based on start date of the channel numbering. Does that happen elsewhere already?

File: atlas-api/src/main/java/org/atlasapi/output/annotation/ChannelAdvertisedAnnotation.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        DateTime advertiseFrom = entity.getAdvertiseFrom();
+        if (advertiseFrom != null && isInFuture(advertiseFrom)) {
```
Comment (tfm) [inline atlas-api/src/main/java/org/atlasapi/output/annotation/ChannelAdvertisedAnnotation.java]: This is incorrect, should be

`if (advertiseFrom == null || !isInFuture(advertiseFrom))`

**Comment by tfm** [comment]
No tests

**Comment by Dalamar42** [comment]
Update the LegacyChannelTransformer test to check you are correctly transforming the new field.

**Comment by Dalamar42** [comment]
See if ChannelWriter has a test class. If it does add a test there. Otherwise LGTM

**Pull request merged**